timelock-impl/src/main/java/com/palantir/atlasdb/timelock/paxos/BatchPaxosLearner.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock.paxos;
+
+import java.util.Map;
+import java.util.Set;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import com.google.common.collect.SetMultimap;
+import com.palantir.paxos.PaxosLearner;
+import com.palantir.paxos.PaxosValue;
+
+@Path("/batch/learner")
+public interface BatchPaxosLearner {
+
+    /**
+     * Batch counterpart to {@link PaxosLearner#learn}. For a given {@link Client} on paxos instance {@code seq},
+     * the learner learns the given {@link PaxosValue}.
+     * <p>
+     * @param paxosValuesByClientAndSeq {@link ClientAndSeq} identifies the client and the instance of paxos that is
+     * being taught; {@link PaxosValue} - for the above {@link ClientAndSeq} - is value being taught.
+     */
+    @POST
+    @Path("learn")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_JSON)
+    void learn(Map<ClientAndSeq, PaxosValue> paxosValuesByClientAndSeq);
```

Comment:
Not actionable: reading this was initially a bit strange to me (a `PaxosValue` already contains the `seq`), but it looks like we already do this in one-client `PaxosLearner` anyway so it seems fair to maintain consistency.

Reply:
Hmm, just realised that I can't serialise a map of `ClientAndSeq` since it'd make it a complex key *weeps. Will have to change it to `SetMultimap<Client, SeqAndPaxosValue>` or as you said, just `SetMultimap<Client, PaxosValue>` since it already contains the sequence.

I'm going to make a class as follows:
```java
@Value.Immutable
interface Seq<T> {
  long seq();
  T value();
}
```

since we use it in a few places.

Reply:
With the above change where I can't have complex map keys, it might make sense to just have a `SetMultimap<Client, PaxosValue>`. I suppose it makes a bit more sense anyway: not sure if there is validation that `seq` and `paxosValue.seq` is the same.